visio_ml/src/modules/compile_results.py

```diff
@@ -0,0 +1,49 @@
+import cv2
+import firebase_admin
+import datetime
+
+class ResultCompiler:
+    PATH_TO_CREDENTIALS = "" # TODO: Add file path to firebase credentials
+
+    # results_db_path: URL to the realtime database for results
+    # results_destination_path: destination file in firebase storage for results
+    # video_db_name: name of the realtime database for videos
+    def __init__(self, exercise, body_part, results_db_path, results_destination_path, video_db_name, frame_rate):
+        self.cred = firebase_admin.credentials.Certificate(self.PATH_TO_CREDENTIALS)
+        self.exercise = exercise
+        self.body_part = body_part
+        self.results_db_path = results_db_path
+        self.results_destination_path = results_destination_path
+        self.video_db_name = video_db_name
+        self.results = []
+        self.total = 0
+        self.video_file_name = f"{exercise}_{body_part}_{frame_rate}fps_{datetime.now()}.avi"
+        self.out = cv2.VideoWriter(self.video_file_name, cv2.VideoWriter_fourcc('M','J','P','G'), frame_rate, (1152, 648))
+    
+    def record_angle(self, angle) -> None:
+        self.results.append(angle)
+        self.total += angle
+
+    def record_frame(self, frame) -> None:
+        self.out.write(frame)
+
+    def get_average_range(self) -> float:
```

Comment:
this is calculating the average angle, but we're looking for the average range, assuming resting is 0 degree, so it should be average of the max values

Reply:
Ok so is this like what we discussed earlier, let's say that the following arguments are passed to `record_angle` (for now just assume that it only takes a float and not a dictionary):

15
27
54
80
105
115 <- Peak
103
89
75
50
20 <- Valley
36
74
91
105 <- Peak
82

So, would it be correct to record the peaks and valleys (we initialize the first valley as 0 degrees).

Then for `self.results`, which we upload to firebase, we store EVERY angle that was recorded. But for `get_average_range()`, we store the average of all adjacent (peak - valley)?